# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/RTCRtpReceiver.cpp
# Commit: f4d3d1b63c2c
# Full Hash: f4d3d1b63c2ccf3bbcb5bf30792a9781b8f98a09
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2025-08-08 21:23:04
# Description:
#   Bug 1975995 - Fix remote track getSettings after peer connection closed. r=bwc
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D260391
# ==============================================================================

diff -r 82915b38e93e -r f4d3d1b63c2c dom/media/webrtc/jsapi/RTCRtpReceiver.cpp
--- a/dom/media/webrtc/jsapi/RTCRtpReceiver.cpp	Fri Aug 08 15:25:10 2025 +0300
+++ b/dom/media/webrtc/jsapi/RTCRtpReceiver.cpp	Fri Aug 08 12:34:03 2025 +0000
@@ -1143,7 +1143,8 @@
 
 Maybe<gfx::IntSize> RTCRtpReceiver::ReceivingSize() const {
   MOZ_ASSERT(NS_IsMainThread());
-  MOZ_ASSERT(mPipeline->mConduit->type() == MediaSessionConduit::VIDEO);
+  MOZ_ASSERT_IF(mPipeline,
+                mPipeline->mConduit->type() == MediaSessionConduit::VIDEO);
   return mReceivingSize.Ref();
 }
 